mautic/mautic
# AVG rounding in reports removed

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | no
| New feature/enhancement? (use the a.x branch)      | yes
| Deprecations?                          | no
| BC breaks? (use the c.x branch)        | no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

In the Emails Sent Report, we used to group our emails by ID and want to calculate AVG Read, AVG Bounced, AVG Unique clicks, AVG Unsubscribed statistics. By default, AVG is rounded to integer, so in our case, detailed values between 0 and 1 are lost. This PR removes AVG rounding.

#### Steps to test this PR:

1. Create Report with Data Source set to Emails Emails Sent.
2. In Data tab add Columns ID, Name and Group by ID. 
3. In Calculated columns add functions as follows:
Funtion: AVG, Column: Read, 
Funtion: AVG, Column: Bounced, 
Funtion: AVG, Column: Unique clicks, 
Funtion: AVG, Column: Unsubscribed statistics, 
4. Publish report and click Save & Close.
5. Check if results in AVG columns are float, not integer. 


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11273"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
